code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -111,13 +111,21 @@
 		update_inv_gloves()
 	else if(I == glasses)
 		glasses = null
+		var/obj/item/clothing/glasses/G = I
+		if(G.tint)
+			update_tint()
+		if(G.prescription)
+			clear_fullscreen("nearsighted")

```

Comment:
This should probably only clear nearsightedness if they have that disability/mutation--just in case there's something else that causes the same overlay, but glasses wouldn't correct for it (ie: a chemical, welding overlay, etc).